src/components/Quiz.js

```diff
@@ -18,19 +18,63 @@ const Quiz = () => {
     setQuestions(quizData);
   }, [quizName]);
 
-  const handleAnswerSelect = e => {
-    const selectedAnswer = { [e.target.name]: Number(e.target.value) };
+  const handleCheckboxAnswers = (e, answers) => {
```

Comment:
A few comments on the logic in this function:

- It looks like `currentQuestionId` is not necessary so we can remove it. It's getting the question ID but we already know that from `e.target.name`
- To calculate `isPreviouslySelected` we can use `includes` rather than `filter` which is a more appropriate method

Reply:
- The name I used `currentQuestionId` is I think misleading. If I called `hasQuestionSelectedAnswer` or `isThisQHasSelectedAnswer`would give more sense I think. I tried removing it but the logic not going to work. For the logic to work I think it is necessary because I used it as reference if a question has selected answer and based on that to remove (unchecked)  or select (checked) answers for the checkbox `checkedAnswer`. 
I am not sure if I expressed myself clearly and correctly. :) 

- I changed the `filter` with `includes`

Reply:
> I used it as reference if a question has selected answer and based on that to remove (unchecked) or select (checked) answers for the checkbox

It looks to me like `isPreviouslySelected` does this already so we don't need `currentQuestionId`. With `isPreviouslySelected` we know whether the user has already selected the answer, and from there we can either remove or select the answer. On my local machine I was able to remove `currentQuestionId` and still have it working so it is possible! If you're still stuck then I can give more hints 👍